### PR TITLE
+define command

### PR DESCRIPTION
**Pull request by QkeleQ10**
I tried adding this command, but it wasn't tested (because you made it hard and didn't document it for us ree) so it probably misses an awful lot. It's not a life changer, so I get it if you choose not to accept my proposed changes.

Please test!

### CODE REVIEW

**Comment by marzeq** [comment]
> Please test!

I can test

File: commands/define.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    usage: "+define <query>",
+    cooldown: 10,
+    allowDM: true,
+    channelWhitelist: ["549894938712866816", "624881429834366986", "730042612647723058", "749391414600925335", "551693960913879071"], // bots staff-bots bot-development bot-translators admin-bots
+    execute(message, strings, args) {
+        message.channel.startTyping()
+
+        var arg = args.join(' ')
+        var xmlhttp = new XMLHttpRequest()
+        var url = `https://www.dictionaryapi.com/api/v3/references/collegiate/json/${arg}?key=${process.env.WEBSTER_COLLEGIATE_KEY}`
```
Comment (ImRodry) [inline commands/define.js:16]: Not sure what `WEBSTER_COLLEGIATE_KEY` is meant to be and we don't have that on .env

File: commands/define.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+module.exports = {
+    name: "define",
+    description: "Gives you the definition of a word in English along with some details.",
+    usage: "+define <query>",
+    cooldown: 10,
+    allowDM: true,
+    channelWhitelist: ["549894938712866816", "624881429834366986", "730042612647723058", "749391414600925335", "551693960913879071"], // bots staff-bots bot-development bot-translators admin-bots
+    execute(message, strings, args) {
+        message.channel.startTyping()
```
Comment (ImRodry) [inline commands/define.js:12]: This should be on the function since that may not run in case there is an error, in which case the bot will be stuck typing

**Review by ImRodry — CHANGES_REQUESTED** [review]
Overall the command looks good but I will need these things changed before testing

File: commands/define.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
+const { errorColor, successColor } = require("../config.json")
+const Discord = require("discord.js")
+
+module.exports = {
+    name: "define",
+    description: "Gives you the definition of a word in English along with some details.",
+    usage: "+define <query>",
+    cooldown: 10,
+    allowDM: true,
```
Comment (ImRodry) [inline commands/define.js:9]: ``
```suggestion
    allowDM: true,
    dev: true,
```

File: commands/define.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    channelWhitelist: ["549894938712866816", "624881429834366986", "730042612647723058", "749391414600925335", "551693960913879071"], // bots staff-bots bot-development bot-translators admin-bots
+    execute(message, strings, args) {
+        message.channel.startTyping()
+
+        var arg = args.join(' ')
+        var xmlhttp = new XMLHttpRequest()
```
Comment (ImRodry) [inline commands/define.js:15]: XMLHttpRequest is not defined. Please test your code, couldn't get this to work

**Review by ImRodry — CHANGES_REQUESTED** [review]

File: commands/define.js (hunk 4 above)
Comment (QkeleQ10) [reply commands/define.js]: I have defined it explicitly now. If it still doesn't work, try running `npm install xmlhttprequest`

File: commands/define.js (hunk 2 above)
Comment (QkeleQ10) [reply commands/define.js:14]: This should also do the trick.

**Comment by ImRodry** [comment]
The API is quite messy and inconsistent. It would take a lot of time and spaghetti code to get this fully working in an understandable manner. The idea overall is very good but I don't feel like we could use this API. This did give me an idea to create a command to translate words though which I will look into in the future

**Pull request closed without merging**